### ISSUE TITLE
Need metadocumentation for :obj: and :func: syntax

### ISSUE THREAD

**Issue by simonster** [JuliaLang/julia#18347]
In the docs, there are links with syntax like:

```
[`RemoteChannel`](:obj:`RemoteChannel`)
[`parse`](:func:`parse`)
```

This syntax should be documented in CONTRIBUTING.md (if this is specific to Julia's docs) or in the manual with a link from CONTRIBUTING.md (if this is something explicitly supported in the docsystem).

**Comment by stevengj** [comment]
This is specific to the [rst](http://docutils.sourceforge.net/rst.html) syntax used for the Julia manual, and is not a general feature of the docstrings AFAIK.  That is, as far as the docsystem knows, `[foo](link)` is just a generic link, where `link` is usually a URL.

That being said, it would be good to have a general way to do cross references in the docsystem, but I'd prefer a syntax that's not tied to rst.

**Comment by kshyatt** [comment]
Seems like http://docs.julialang.org/en/latest/manual/documentation/ would be a good place for this?

**Comment by tkelman** [comment]
Yes, though it's worth double-checking with @MichaelHatherly if there's a nicer-looking cross-reference syntax planned (or implemented already?) in Documenter.jl which we'll likely transition to. The current format might just be an intermediate state to make the genstdlib partially-autogenerated-rst situation work.

**Comment by MichaelHatherly** [comment]
It's just an intermediate state until we move to markdown docs completely. Docstring cross references in Documenter are just

```
[`foo`](@ref)
```

and section/header cross references are

```
[Title](@ref)
```

**Comment by MichaelHatherly** [comment]
Closed by #18585. Cross-reference syntax is documented [here](https://juliadocs.github.io/Documenter.jl/stable/man/syntax.html#@ref-link-1).
